src/main.rs

```diff
@@ -1,28 +1,57 @@
+use std::io::{Stdin, Stdout, Write};
+
 fn main() {
     println!("Olá! 😃");
 
-    loop {        
-        println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
+    loop {
+       let mut ask_todo = Terminal::new();
+       let todo = ask_todo.ask_for_new_todo();
 
-        if input() == "s" {
-            create_todo();
-        } else {           
-            println!("\nTodo list finalizado! 🤠");
-            break;
-        }
+       ask_todo.show_todo(&todo);
     }
 }
 
-fn create_todo() {
-    
-    println!("\nQual TODO deseja criar?");
-    let todo = input();
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String
+}
 
-    println!("\n✅: {todo}");
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
 }
 
-fn input() -> String {
-    let mut buf = String::new();
-    std::io::stdin().read_line(&mut buf).unwrap();
-    buf.trim().to_string()
- }
+impl Terminal {
+    fn new() -> Self {
+
+        Terminal {
+            stdin: std::io::stdin(), 
+            stdout: std::io::stdout() 
+        }
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {        
+
+        println!("\nVocê gostaria de adicionar um novo TODO? 🤔 (Digite: 's' para SIM ou qualquer outra tecla para NÃO)");
+    
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();               
+        
+        if buf.trim().to_string() == "s" {
```

Comment:
Como o método `trim` retorna uma `&str`, você não precisa usar o `.to_string()` para convertê-lo em `String`. :)

```suggestion
        if buf.trim() == "s" {
```

Numa seção futura você vai aprender a diferença entre `&str` e `String`.

Reply:
Belezinha.